src/Cake.Svn/Add/SvnAdder.cs

```diff
@@ -0,0 +1,45 @@
+﻿using System;
+using Cake.Core;
+using Cake.Svn.Internal.Extensions;
+
+namespace Cake.Svn.Add
+{
+    /// <summary>
+    /// Class for adding files or directories to Subversion directory trees.
+    /// </summary>
+    public sealed class SvnAdder : SvnTool<SvnAddSettings>
+    {
+        private readonly ICakeEnvironment _environment;
+
+        /// <summary>
+        /// Initializes a new instance of the <see cref="SvnAdder"/> class.
+        /// </summary>
+        /// <param name="environment">The Cake environment.</param>
+        /// <param name="clientFactoryMethod">Method to use to initialize a Subversion client.</param>
+        public SvnAdder(ICakeEnvironment environment, Func<ISvnClient> clientFactoryMethod)
+            : base(clientFactoryMethod)
+        {
+            environment.NotNull(nameof(environment));
+            clientFactoryMethod.NotNull(nameof(clientFactoryMethod));
+
+            _environment = environment;
+        }
+
+        /// <summary>
+        /// Add a file or directory to Subversion.
+        /// </summary>
+        /// <param name="fileOrDirectoryToAdd">The filename or directoryname to add.</param>
+        /// <param name="settings">The settings.</param>
+        /// <returns>True if the command were successful. Otherwise False will be returned.</returns>
+        public bool Add(string fileOrDirectoryToAdd, SvnAddSettings settings)
```

Comment:
Cake has [FilePath](https://cakebuild.net/api/Cake.Core.IO/FilePath/) and [DirectoryPath](https://cakebuild.net/api/Cake.Core.IO/DirectoryPath/) types which should be used for passing file or directory paths. Since file and paths are supported this will require an overload of the method.

Also you should think what kind of paths are supported and handle them accordingly. Only absolute paths? Also relative paths? IMHO both should be possible (see https://github.com/cake-contrib/Cake.Svn/blob/develop/src/Cake.Svn/Export/SvnExporter.cs#L44 for how this is handled for svn export).